addons/web/static/src/views/graph/graph_renderer.js

```diff
@@ -15,7 +22,102 @@ import { ReportViewMeasures } from "@web/views/view_components/report_view_measu
+ * see this last one correctly.
+ */
+const gridOnTop = {
+    id: "gridOnTop",
+    afterDraw: (chart) => {
+        const elements = chart.getDatasetMeta(0).data || [];
+        const ctx = chart.ctx;
+        const chartArea = chart.chartArea;
+        const yAxis = chart.scales.y;
+        const xAxis = chart.scales.x;
+
+        ctx.lineWidth = 1;
+        ctx.strokeStyle = GRAPH_GRID_COLOR;
+
+        // Draw Y axis scale
+        yAxis.ticks.forEach((value, index) => {
+            const y = yAxis.getPixelForTick(index);
+            ctx.beginPath();
+            // Draw the line scale
+            ctx.moveTo(chartArea.left, y);
+            ctx.lineTo(chartArea.right, y);
+            // Draw the tick mark
+            ctx.moveTo(chartArea.left - 8, y);
+            ctx.lineTo(chartArea.left, y);
+            ctx.setLineDash([]);
+            ctx.stroke();
+        });
+
+        // Draw X axis tick marks
+        xAxis.ticks.forEach((value, tickIndex) => {
+            const x = xAxis.getPixelForTick(tickIndex);
+            ctx.beginPath();
+            ctx.moveTo(x, chartArea.bottom);
+            ctx.lineTo(x, chartArea.bottom + 8);
+            ctx.stroke();
+        });
+
+        // Draw the X axis dashed line
+        elements.forEach((point, eltIndex) => {
+            xAxis.ticks.forEach((value, tickIndex) => {
+                if (point.active && eltIndex === tickIndex) {
+                    const x = xAxis.getPixelForTick(tickIndex);
+                    ctx.beginPath();
+                    ctx.moveTo(x, chartArea.top);
+                    ctx.lineTo(x, chartArea.bottom);
+                    ctx.strokeStyle = GRAPH_GRID_COLOR;
+                    ctx.stroke();
+                }
+            });
+        });
+    },
+};
+
+/**
+ * Used to generate the max value of the grid (line & bar charts).
+ * The purpose is to keep a bit of space between the highest data
+ * and the top of the grid.
+ * @param {Array} data
+ * @param {Boolean} isStacked
+ * @returns {Number}
+ */
+function getMaxValue(data, isStacked) {
+    let max = 0;
+    // If there is data
+    if (data.datasets.length) {
+        if (isStacked) {
+            // Initialize an array to define the addition of values according to their index
+            const maxValues = [];
+            // Add values for each index
+            data.datasets.forEach((dataset) => {
+                dataset.data.forEach((value, index) => {
+                    maxValues[index] = (maxValues[index] || 0) + value;
+                });
+            });
+            // Get the highest value
+            max = Math.max(...maxValues);
+        } else {
+            data.datasets.forEach((dataset) => {
+                const datasetMax = Math.max(...dataset.data);
+                if (datasetMax > max) {
+                    max = datasetMax;
+                }
+            });
+        }
+        // Return the max value and increase it a bit
+        return 1.1 * max;
```

Comment:
I think we should consider the sign of max. If it is negative, you would like to multiply it by 0.9.
We should maybe add a also a suggestedMin with same thing about factor but 1.1 and 0.9 exchanged